modules/extended_tags.py

```diff
@@ -0,0 +1,58 @@
+"""
+    modules.extended_tags
+    ~~~~~~~~~~~~~~~~~~~~
+
+    Gives more power to tagging.
+"""
+
+from collections import Counter
+from collections import defaultdict
+
+
+def relatedness(taglist):
+    related = defaultdict(lambda: defaultdict(int))
+    for tag_set in taglist:
+        for tag in tag_set:
+            related_tags = tag_set.symmetric_difference({tag})
+            for other_tag in related_tags:
+                related[tag][other_tag] += 1
+    return related
```

Comment:
It took me a few minutes of staring at this to figure out what it's doing. Docstrings for these functions would be nice.